frontend/src/components/Review/ReviewTagList/ReviewTagList.tsx

```diff
@@ -1,28 +1,38 @@
-import { Button, Heading } from '@fun-eat/design-system';
+import { Button, Heading, Spacing } from '@fun-eat/design-system';
 import { useState } from 'react';
 import styled from 'styled-components';
 
 import ReviewTagItem from '../ReviewTagItem/ReviewTagItem';
 
 import { SvgIcon } from '@/components/Common';
+import { TAG_TITLE } from '@/constants';
 import reviewTagList from '@/mocks/data/reviewTagList.json';
+import { isTagNameVariant, type TagNameOption } from '@/types/common';
 
-const MAX_DISPLAYED_TAGS = 8;
-const MAX_SELECTED_TAGS = 3;
+const MAX_DISPLAYED_TAGS = 3;
 
 const ReviewTagList = () => {
   const [maxDisplayedTags, setMaxDisplayedTags] = useState(MAX_DISPLAYED_TAGS);
   const [selectedTags, setSelectedTags] = useState<number[]>([]);
 
-  const canShowMore = reviewTagList.length > maxDisplayedTags;
-  const displayedTagList = reviewTagList.slice(0, maxDisplayedTags);
+  const convertTagName = (tagName: TagNameOption): string => TAG_TITLE[tagName] || null;
+
+  const getMaxTagsInGroup = (tagList: typeof reviewTagList) => {
+    return tagList.reduce((max, currentGroup) => {
+      return Math.max(max, currentGroup.tags.length);
+    }, 0);
+  };
+
+  const canShowMore = maxDisplayedTags < getMaxTagsInGroup(reviewTagList);
 
   const showMoreTags = () => {
-    setMaxDisplayedTags(maxDisplayedTags + MAX_DISPLAYED_TAGS);
+    setMaxDisplayedTags(getMaxTagsInGroup(reviewTagList));
   };
 
-  const toggleTagSelection = (id: number, isSelected: boolean) => {
-    if (selectedTags.length >= MAX_SELECTED_TAGS && !isSelected) {
+  const toggleTagSelection = (id: number) => {
+    const isSelected = selectedTags.includes(id);
```

Comment:
```suggestion
  const toggleTagSelection = (id: number, isSelected: boolean) => {
```
props 타입과 맞춘다면 이렇게 되겠네요